src/cpp/ext/otel/otel_plugin.cc

```diff
@@ -175,12 +186,21 @@ void OpenTelemetryPluginBuilder::BuildAndRegisterGlobal() {
   grpc_core::ServerCallTracerFactory::RegisterGlobal(
       new grpc::internal::OpenTelemetryServerCallTracerFactory);
   grpc_core::CoreConfiguration::RegisterBuilder(
-      [](grpc_core::CoreConfiguration::Builder* builder) {
+      [channel_selector = std::move(channel_selector_)](
+          grpc_core::CoreConfiguration::Builder* builder) mutable {
         builder->channel_init()->RegisterStage(
             GRPC_CLIENT_CHANNEL, /*priority=*/INT_MAX,
-            [](grpc_core::ChannelStackBuilder* builder) {
-              builder->PrependFilter(
-                  &grpc::internal::OpenTelemetryClientFilter::kFilter);
+            [channel_selector = std::move(channel_selector)](
+                grpc_core::ChannelStackBuilder* builder) {
+              // Only register the filter if no channel selector has been set or
+              // the channel selector returns true for the target.
+              if (!channel_selector ||
```

Comment:
Prefer to write this as `channel_selector == nullptr`.

Reply:
Done